### PR TITLE
Enable async entity transforms

### PR DESCRIPTION
**Pull request by felixjung**
This is a first attempt at addressing #87.

### CODE REVIEW

File: lib/transform/transform-space.js [hunk 1]
```diff
@@ -11,22 +12,30 @@ const spaceEntities = [
...
+    return Promise.map(sourceEntities, (entity) => typeTransform(entity))
       .then((entities) => {
         newSpace[type] = entities
         return newSpace
       })
   }, newSpace)
 }
+
+function applyTransform (entity, transform, destinationEntities, destinationSpace) {
+  return Promise.resolve(transform(entity, destinationEntities, destinationSpace))
```
Comment (Khaledgarbaya) [inline lib/transform/transform-space.js]: Maybe you can do

```js
return Promise.props({original: entity, transformed: transform(entity, destinationEntities, destinationSpace)})
```
so you can skip that extra `then`

Comment (felixjung) [inline lib/transform/transform-space.js]: Ah, thanks. Makes total sense.

**Comment by felixjung** [comment]
Sorry, this took a while. After some time I though I would not need this. Business requirements have changed, now I do 😂

**Comment by Khaledgarbaya** [comment]
@felixjung turns out this change has a runtime error complaining about _partialRight not a function

**Comment by felixjung** [comment]
Oupsis :D Not good!

**Pull request merged**